homeassistant/components/moehlenhoff_alpha2/climate.py

```diff
@@ -0,0 +1,181 @@
+
+        try:
+            await self.coordinator.async_set_target_temperature(
+                self.coordinator.data[self.heatarea_id]["ID"], target_temperature
+            )
+            self.coordinator.data[self.heatarea_id]["T_TARGET"] = target_temperature
+        except aiohttp.web.HTTPException as http_err:
+            raise HomeAssistantError(
+                "Failed to set target temperature, communication error with alpha2 base"
+            ) from http_err
+
+    @property
+    def preset_mode(self) -> str:
+        """Return the current preset mode."""
+        if self.coordinator.data[self.heatarea_id]["HEATAREA_MODE"] == 1:
+            return PRESET_DAY
+        if self.coordinator.data[self.heatarea_id]["HEATAREA_MODE"] == 2:
+            return PRESET_NIGHT
+        return PRESET_AUTO
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new operation mode."""
+        heatarea_mode = 0
+        if preset_mode == PRESET_DAY:
+            heatarea_mode = 1
+        elif preset_mode == PRESET_NIGHT:
+            heatarea_mode = 2
+
+        try:
+            await self.coordinator.async_set_heatarea_mode(
+                self.coordinator.data[self.heatarea_id]["ID"], heatarea_mode
+            )
+        except aiohttp.web.HTTPError as http_err:
+            raise HomeAssistantError(
+                "Failed to set preset mode, communication error with alpha2 base"
+            ) from http_err
+        else:
+            self.coordinator.data[self.heatarea_id]["HEATAREA_MODE"] = heatarea_mode
+            if heatarea_mode == 1:
+                if self.coordinator.get_cooling():
+                    self.coordinator.data[self.heatarea_id][
+                        "T_TARGET"
+                    ] = self.coordinator.data[self.heatarea_id]["T_COOL_DAY"]
+                else:
+                    self.coordinator.data[self.heatarea_id][
+                        "T_TARGET"
+                    ] = self.coordinator.data[self.heatarea_id]["T_HEAT_DAY"]
+            elif heatarea_mode == 2:
+                if self.coordinator.get_cooling():
+                    self.coordinator.data[self.heatarea_id][
+                        "T_TARGET"
+                    ] = self.coordinator.data[self.heatarea_id]["T_COOL_NIGHT"]
+                else:
+                    self.coordinator.data[self.heatarea_id][
+                        "T_TARGET"
+                    ] = self.coordinator.data[self.heatarea_id]["T_HEAT_NIGHT"]
+
+    @property
+    def extra_state_attributes(self):
+        """Return the optional state attributes."""
+        data = super().extra_state_attributes or {}
+        data["islocked"] = self.coordinator.data[self.heatarea_id].get(
+            "ISLOCKED", False
+        )
+        data["lock_available"] = self.coordinator.data[self.heatarea_id].get(
+            "LOCK_AVAILABLE", False
+        )
+        data["block_hc"] = self.coordinator.data[self.heatarea_id].get(
```

Comment:
Please explain what these values in the extra state attributes represent.

Reply:
is that ok?